### PR TITLE
ckb/ckb-next: 0.2.9 -> 0.3.2, and cleanup

### PR DESCRIPTION
**Pull request by kierdavis**
* updated ckb package from 0.2.9 to 0.3.0 (EDIT: now 0.3.2)
* renamed package and module from "ckb" to "ckb-next", following the move by the upstream developers to do the same
* minor tidy-ups

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1768,7 +1768,8 @@ with pkgs;
 
   checkbashisms = callPackage ../development/tools/misc/checkbashisms { };
 
-  ckb = libsForQt5.callPackage ../tools/misc/ckb { };
+  ckb-next = libsForQt5.callPackage ../tools/misc/ckb-next { };
+  ckb = ckb-next; # alias retained for backwards compatibility
```
Comment (samueldr) [inline pkgs/top-level/all-packages.nix]: Please move the alias to `pkgs/top-level/aliases.nix`.

(A bunch of aliases are marked as `TODO: move to aliases.nix` in this file.)

File: pkgs/tools/misc/ckb-next/default.nix [hunk 2]
```diff
@@ -1,14 +1,14 @@
-{ stdenv, fetchFromGitHub, substituteAll, libudev, pkgconfig, qtbase, qmake, zlib, kmod }:
+{ stdenv, fetchFromGitHub, substituteAll, libudev, pkgconfig, qtbase, cmake, zlib, kmod }:
 
 stdenv.mkDerivation rec {
-  version = "0.2.9";
+  version = "0.3.0";
```
Comment (samueldr) [inline pkgs/tools/misc/ckb-next/default.nix]: `0.3.1` was released meanwhile, if it isn't too much, update and verify it still works (assuming you have the hardware.)

**Review by samueldr — CHANGES_REQUESTED** [review]
Thanks!

Minor changes requested, see the comments. Other than those nitpicks it looks like an approval from me.

I would additionally like if you could add the option rename to the release-notes (`nixos/doc/manual/release-notes/rl-1809.xml`). If you can't or don't want to write the release notes just ask and I'll handle it.

**Comment by kierdavis** [comment]
@samueldr I've pushed an updated version of this PR. Changes are:

* Upgraded to version 0.3.2
* Rebased onto master to fix merge conflicts
* Moved alias to `pkgs/top-level/aliases.nix` as requested
* Added an entry to the 19.03 release notes

P.S. Github's UI appears to be behaving weirdly for me: on some page loads, the head commit of `kierdavis/ckb-update-and-cleanup` is wrong, resulted in outdated logs/diffs being shown as if they are current. When reviewing please double-check that the most recent commit on the branch is `7a0d8496537a8b31a0697a8d08fa042837fb467c`.

File: nixos/modules/hardware/ckb-next.nix [hunk 3]
```diff
@@ -3,17 +3,22 @@
...
-  cfg = config.hardware.ckb;
+  cfg = config.hardware.ckb-next;
 
 in
   {
-    options.hardware.ckb = {
+    imports = [
+      (mkRenamedOptionModule ["hardware" "ckb" "enable"] ["hardware" "ckb-next" "enable"])
+      (mkRenamedOptionModule ["hardware" "ckb" "package"] ["hardware" "ckb-next" "package"])
+    ];
```
Comment (samueldr) [inline nixos/modules/hardware/ckb-next.nix]: Oops, looks like I missed those renames. I personally don't mind if those are kept local to the relevant module, but it looks like most renames are in `nixos/modules/rename.nix`.

I wouldn't hold back the change on account of that, though let's say what you yourself think about it, and if others have a strong opinion about that.

**Review by samueldr — APPROVED** [review]
Other than the really nit-picky comment, and the merge conflict, this looks fine. Everything fixed as asked.

**Comment by kierdavis** [comment]
@samueldr Thanks for the review! I've fixed the merge conflict and moved the option renames as suggested.

**Comment by kierdavis** [comment]
Thanks!

**Pull request merged**